dashboard/app/models/user_permission.rb

```diff
@@ -44,6 +44,9 @@ class UserPermission < ApplicationRecord
     PROGRAM_MANAGER = 'program_manager'.freeze,
     # Grants ability to be the instructor of any course no matter instructor_audience
     UNIVERSAL_INSTRUCTOR = 'universal_instructor'.freeze,
+    # Grants access to use of AI API
+    AI_API_ACCESS = 'ai_api_access'.freeze,
```

Comment:
Nit: I feel like "access" is redundant in naming a permission...maybe just `AI_API` (or `AI_CHAT` if its possible we end up with different access to different AI APIs)?

Reply:
We recently removed a user permission type named `hidden_script_access` ([Slack thread](https://codedotorg.slack.com/archives/C045UAX4WKH/p1683807553517929?thread_ts=1683741668.115499&cid=C045UAX4WKH)) so I named this similarly. But I do like `AI_CHAT_ACCESS`. What do you thnink?

Reply:
Good with me!